qiita_db/portal.py

```diff
@@ -5,13 +5,66 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+import warnings
 
 from .sql_connection import SQLConnectionHandler
 from .util import convert_to_id
+from .exceptions import QiitaDBError, QiitaDBDuplicateError
 from qiita_core.qiita_settings import qiita_config
+from qiita_core.exceptions import IncompetentQiitaDeveloperError
 
 # Only make functions available for main portal
 if qiita_config.portal == "QIITA":
+    def create_portal(portal, desc):
```

Comment:
I agree with both of you. The encapsulation of these methods under a namespace makes sense, however there doesn't appear to be much common state so they'd likely end up being `staticmethod`s. If the object is entirely composed of `staticmethod`s, then it probably shouldn't be an object. 

The bigger issue I see is that these methods are optionally present in the namespace. The current approach will require that any caller of, say, `create_portal` will first have to either interrogate the `globals` to see if the method exists or check against `qiita_config.portal`. Even then, this is fragile as imports are not assured to work:

```python
try:
    from qiita_db.portal import create_portal
except ImportError:
    create_portal = lambda *args: None
``` 

Why does `qiita_db` care what portal is specified in the config when doing things like `create_portal`? Are the portals nested? If so, then I apologize for missing that design goal. If they aren't, then really I think the consideration should be whether an administrator is performing the action.